src/components/TextInputs/__tests__/PasswordTextInput.test.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+import { TextInput as RNTextInput, TouchableOpacity } from 'react-native'
+import renderer from 'react-test-renderer'
+import PasswordTextInput from '../PasswordTextInput'
+
+describe('PasswordTextInput', () => {
+  describe('when focusable is true', () => {
+    it('renders with white border when on blur', () => {
+      const testRenderer = renderer.create(<PasswordTextInput focusable />)
+      const input = testRenderer.root.findByType(RNTextInput)
+      input.props.onBlur()
+      expect(testRenderer.toJSON()).toMatchSnapshot()
+    })
+
+    it('renders with green border when on focus', () => {
+      const testRenderer = renderer.create(<PasswordTextInput focusable />)
+      const input = testRenderer.root.findByType(RNTextInput)
+      input.props.onFocus()
+      expect(testRenderer.toJSON()).toMatchSnapshot()
+    })
+  })
+
+  describe('when focusable is false', () => {
+    it('renders with white border when on blur', () => {
+      const testRenderer = renderer.create(<PasswordTextInput focusable={false} />)
+      const input = testRenderer.root.findByType(RNTextInput)
+      input.props.onBlur()
+      expect(testRenderer.toJSON()).toMatchSnapshot()
+    })
+
+    it('also renders with white border when on focus', () => {
+      const testRenderer = renderer.create(<PasswordTextInput focusable={false} />)
+      const input = testRenderer.root.findByType(RNTextInput)
+      input.props.onFocus()
+      expect(testRenderer.toJSON()).toMatchSnapshot()
+    })
+  })
+
+  describe('when password toggle is clicked', () => {
+    it('correctly sets passwordVisible state', () => {
```

Comment:
Maybe instead of checking the state, we can check if the image being displayed is the correct one. That way, we don't rely on the internals of `state` and focus on the feature itself. That would be more black box instead of white box testing.

And by doing that, we could have 2 tests: one checking for the `visible` icon and another checking for the `invisible` icon.

Reply:
For ref on how to find an Image and check for its source:

https://github.com/loadsmart/blocks/blob/38185b1bc13bcd1316173758b0087dc8ca027eee/src/components/Buttons/__tests__/PrimaryButton.test.tsx#L24-L25

Reply:
@barbosa Sure!